setup.py

```diff
@@ -0,0 +1,36 @@
+from setuptools import setup, find_namespace_packages
+from pathlib import Path
+
+HERE = Path(__file__).parent
+
+script_names = [
+    path.with_suffix("").name
+    for path in (HERE / 'scanpy_scripts').glob('scanpy_*.py')
+]
+scripts = [
+    f'{name.replace("_", "-")}=scanpy_scripts.{name}:main'
+    for name in script_names
+]
+
+setup(
+    name='scanpy-scripts',
+    version='1.0',
+    description='Scripts for using scanpy from the command line',
+    author='nh3',
+    #author_email='',
+    url='https://github.com/ebi-gene-expression-group/scanpy-scripts',
+    packages=find_namespace_packages(),
+    scripts=[
+        'scanpy-scripts-tests.sh',
+        'scanpy-scripts-tests.bats',
+    ],
+    entry_points=dict(
+        console_scripts=scripts,
+    ),
+    install_requires=[
+        'matplotlib',
+        'pandas',
+        'scanpy',
```

Comment:
OK, sure! Do you know why this error happens? Then we can fix it instead of depending on an old scanpy version.

Or fix it in scanpy. A breaking change between 1.3.2 and 1.3.3 seems like a bug!

Reply:
It happens when calling `pl.rank_genes_groups_stacked_violin()` on an object without saved raw data and leaving `use_raw` to its default value `None`. `stacked_violin()` in `anndata.py` doesn't set `use_raw` to `False` if initially `None` and `adata.raw` is not present and passes it to `_compute_dendrogram()`, which tries to access `adata.raw` if `use_raw` is not `False`.